Minor improvements to the custom resources docs

tas50 opened the pull request.

chef-delivery commented:
This PR has passed 'Verify' and is ready for review and approval!
  Use: '@delivery approve' when code review is complete.

jamescott commented:
@delivery approve

chef-delivery commented:
Change: 02119578-8697-43e4-ae76-999a99fd15b7 approved by: @jamescott

chef-delivery commented:
This PR has passed 'Acceptance' and is ready to be delivered!
  Use: '@delivery deliver' when validated in acceptance.

jamescott commented:
@delivery deliver

chef-delivery commented:
Change: 02119578-8697-43e4-ae76-999a99fd15b7 delivered by: @jamescott

tas50 committed:
Put the version requirements after the description of what a custom resource is

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Clarify that we're loading state here not the actual properties

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Remove a load_current_value block that's not used

This wouldn't be necessary in this resource

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Remove action_class.class_eval references

Turns out we don't want to do this

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Reference property names with new_resource in examples

This will prevent people from being burned by weird ruby errors when they have name collisions

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Add custom resources to the concept menu

It's pretty core to Chef

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Copy tags to other pages

Signed-off-by: Tim Smith <tsmith@chef.io>

davidwrede commented on chef_master/source/custom_resources.rst:
```diff
@@ -21,14 +16,18 @@ For example, Chef includes built-in resources to manage files, packages, templat
 
 .. end_tag
 
+Custom resources were introduced in Chef version 12.5 and are now the preferred method of writing your own resources in Chef. If you are using an older version of the chef-client, please use the version picker (in the top left of the navigation) to select your version, and then choose the same topic from the navigation tree ("Extend Chef > Custom Resources"). See also https://github.com/chef-cook ...
```
We no longer have the version picker in the main doc set; however, you can refer people off to the archive site for the older docs. The "Custom Resources" topic for the Chef 12.4 doc set is here: https://docs-archive.chef.io/release/12-4/custom_resources.html

Commit:
link to legacy documentation / cleanup

brewn commented:
![tenor-135995138](https://user-images.githubusercontent.com/21040805/26980650-bb306f28-4d00-11e7-9cf4-bb1cce3fa256.gif)
